### DISCUSSION TITLE
Check Password last changed from user accounts

### DISCUSSION THREAD

**Discussion by xKugeki** [dani-garcia/vaultwarden#4623] [Q&A]
Hey is there a way to check when a user changed his own account password?
Like "last changed.." so that I can inform them to change e.g.

I checked db and there is no entry and in logs one can see that there was a change but not which account changed.
Maybe there is a way to find that out :)

Thanks in advance and have a nice day :D

**Comment by stefan0xC** [comment]
I think the `updated_at` column of the `users` table will be updated on password changes (but this won't be specific only to password changes so it most likely won't be useful).

Better way if you want this would be to enable [event logs](https://bitwarden.com/help/event-logs/) by setting `ORG_EVENTS_ENABLED=true`, so that password changes will be logged in the `event` table.
https://github.com/dani-garcia/vaultwarden/blob/f05398a6b36891e6e979a39937f8f8eaa6360d52/src/api/core/accounts.rs#L365

> in logs one can see that there was a change but not which account changed.

Well, it should be the account which has "logged in successfully" in the web-vault prior to the password change which might make it possible to pin it down. But in general, I don't think it's advisable to inform specific users that they should change their passwords. If you set a password policy all users should update their passwords if necessary but this should be enforced by Vaultwarden (cf #4507).

**Reply by xKugeki** [reply]
Hey @stefan0xC thank you for your answer.
yeah i allready checked last updated but as u said this is not that useful :D 

i enabled event logs and put the remain days to 365 and if i export the .csv then i can filter for recent password changes. Not the coolest way but it works :D is there a way to export the csv via commandline or powershell?

**Reply by stefan0xC** [reply]
No idea. It might be easier to just query the database directly, eg. something like
```SQL
SELECT email FROM users WHERE uuid IN (SELECT user_uuid FROM event WHERE event_type = 1001);
```
